manager/manager.go

```diff
@@ -117,6 +119,42 @@ func (m *Manager) ListInstances(limit int64, tagFilters []*TagFilter) ([]*Instan
 	return out, nil
 }
 
+// ListInstances fetches a list of instances managed by SSM. Paginates until all responses have been collected.
```

Comment:
I think it would be nice if we could rewrite `describeInstances` so that we could use it inside `FilterInstances`, and then in turn use `FilterInstances` inside `ListInstances`. Makes things a little more DRY and ensures that the filter behaviour stays consistent. 

Great if you are up for that, if not I guess we can always change it later 👍

Reply:
Yes, totally agree...